sentry-delayed_job/lib/sentry/delayed_job/plugin.rb

```diff
@@ -12,15 +12,24 @@ class Plugin < ::Delayed::Plugin
         lifecycle.around(:invoke_job) do |job, *args, &block|
           next block.call(job, *args) unless Sentry.initialized?
 
+          
           Sentry.with_scope do |scope|
-            scope.set_contexts(**generate_contexts(job))
+            contexts = generate_contexts(job)
+            scope.set_transaction_name(contexts.dig(ACTIVE_JOB_CONTEXT_KEY, :job_class) || contexts.dig(DELAYED_JOB_CONTEXT_KEY, :job_class))
+            scope.set_contexts(**contexts)
             scope.set_tags("delayed_job.queue" => job.queue, "delayed_job.id" => job.id.to_s)
 
+            transaction = Sentry.start_transaction(name: scope.transaction_name, op: "delayed_job")
+            scope.set_span(transaction) if transaction
+
             begin
               block.call(job, *args)
+
+              finish_transaction(transaction, 200)
             rescue Exception => e
               capture_exception(e, job)
-
+              finish_transaction(transaction, 500)
+              scope.clear
```

Comment:
this is unnecessary. we have this in `sentry-sidekiq` because the middleware can't be wrapped inside a local scope (`Sentry.with_scope`).